sql/17-fix_revision_disorder.sql

```diff
@@ -0,0 +1,157 @@
+--
+-- Change ID of any revision having start_time at a later time than
+-- any other revision with higher ID
+--
+-- New IDs for revisions to be moved will be assigned in
+-- start_time order and start after the highest existing
+-- revision ID.
+--
+-- The number of moved revisions is returned.
+--
+-- {
+CREATE OR REPLACE FUNCTION ver_fix_revision_disorder()
+RETURNS bigint AS
+$FIX$
+DECLARE
+
+  v_rec RECORD;
+  v_newid bigint;
+  v_numversionedtables int;
+  v_totdisordered bigint;
+  v_numdisordered bigint;
+
+BEGIN
+
+-- 1. Make sure sequence is set to stop filling gaps
+
+  PERFORM setval('@extschema@.revision_id_seq',
+    greatest(
+      (select max(id) FROM @extschema@.revision),
+      (select last_value from @extschema@.revision_id_seq)
+    ), true);
+
+-- 2. Prepare update queries
+  v_numversionedtables := 0;
+  FOR v_rec IN SELECT schema_name, table_name
+               FROM table_version.versioned_tables
+  LOOP
+    BEGIN
+      EXECUTE format('PREPARE "p_uc%s" AS '
+          'UPDATE table_version.%s_%s_revision '
+          'SET _revision_created = $1 WHERE _revision_created = $2',
+          v_numversionedtables, v_rec.schema_name, v_rec.table_name);
+      EXECUTE format('PREPARE "p_ue%s" AS '
+          'UPDATE table_version.%s_%s_revision '
+          'SET _revision_created = $1 WHERE _revision_created = $2',
```

Comment:
this was a bug, now filed as https://github.com/linz/postgresql-tableversion/issues/115